src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -778,6 +778,7 @@ private BoundExpression BindTupleExpression(TupleExpressionSyntax node, Diagnost
             if (hasNaturalType)
             {
                 tupleTypeOpt = TupleTypeSymbol.Create(node.Location, elements, locations, elementNamesArray, this.Compilation, node, diagnostics);
+                tupleTypeOpt.CheckAllConstraints(this.Conversions, node.Location, diagnostics);
```

Comment:
They are checked by the caller in method parameters, but that's not the case for expressions; per https://github.com/dotnet/roslyn/pull/15641#discussion_r91232311.